checker/tests/determinism/MapEntrySet.java

```diff
@@ -0,0 +1,20 @@
+import java.util.Map;
+import java.util.Set;
+import org.checkerframework.checker.determinism.qual.Det;
+import org.checkerframework.checker.determinism.qual.NonDet;
+import org.checkerframework.checker.determinism.qual.OrderNonDet;
+
+public class MapEntrySet {
+    void testNDMap(@NonDet Map<@NonDet String, @NonDet String> map) {
+        // :: error: (assignment.type.incompatible)
+        @Det Set<Map.@Det Entry<@NonDet String, @NonDet String>> entries = map.entrySet();
```

Comment:
I think this would fail regardless of  whether the changes in the pull request were present because the top level type of the right side is `@NonDet` and the left side has `@Det`. I think it would be sufficient to test, `@NonDet Set<Map. @NonDet Entry<@NonDet String, @NonDet String>> entries = map.entrySet()` and not expect an error because if the new jdk annotation were not present this assignment would fail.